src/pal/inc/unixasmmacrosarm.inc

```diff
@@ -146,17 +146,46 @@ C_FUNC(\Name\()_End):
 
         __PWTB_StackAlloc = __PWTB_TransitionBlock
 
+	// Make the stack operation compatible with GDB and LIBUNWIND
+	push {r7, lr}
+	mov r7, sp
+	sub sp, #((__PWTB_StackAlloc) + 4 * 4 + 9 * 4)
+
         .ifnc \pushArgRegs, DoNotPushArgRegs
-                PUSH_ARGUMENT_REGISTERS
+		sub r7, #12
+		str r0, [r7]
+		str r1, [r7 , #4]
+		str r2, [r7 , #8]
+		str r3, [r7 , #12]
         .endif
 
-        PUSH_CALLEE_SAVED_REGISTERS
-        
-        alloc_stack     __PWTB_StackAlloc
+	sub r7, #36
+	str r4, [r7]
+	str r5, [r7 , #4]
+	str r6, [r7 , #8]
+	// Store R7 down there.
+	str r8, [r7 , #16]
+	str r9, [r7 , #20]
+	str r10, [r7 , #28]
+	str r11, [r7 , #32]
+	str lr, [r7 , #36]
+
+	// Get R7 from the first push {r7, lr}
+	ldr r6, [sp , #((__PWTB_StackAlloc) + 4 * 4 + 9 * 4 + 4)]
+	str r6, [r7 , #12]
+
+        //alloc_stack     __PWTB_StackAlloc
 
         .if (__PWTB_SaveFPArgs == 1)
-                add r6, sp, #(__PWTB_FloatArgumentRegisters)
-                vstm r6, {d0-d7}
```

Comment:
Because it seems (by few instances of experiments) to confuse GDB and LIBUNWIND. After VSTM and then, BL, GDB thinks that LR of ThePreStub is saved at the place where D7 is stored with VSTM.